compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -165,6 +165,26 @@ private class ExtractDependenciesCollector(rec: DependencyRecorder) extends tpd.
         rec.addClassDependency(parent.tpe.classSymbol, depContext)
     }
 
+  // Only reference DependencyByMacroExpansion if it an be found on the classpath,
+  // as it was added later to the zinc.apiinfo DependencyContext enum
+  // e.g. pre 1.10.x sbt would throw java.lang.NoSuchFieldError errors here
+  lazy val allowsDependencyByMacroExpansion =
+    classOf[DependencyContext].getFields().exists(_.getName() == "DependencyByMacroExpansion")
```

Comment:
I think this is a mistake, this is using the static type (meaning field names are statically available), so this will always pass.

There should be a way to dynamically test actual linked zinc version

Reply:
I've tested just now and something like this:
```scala
  lazy val allowsDependencyByMacroExpansion =
    val reader = new BufferedReader(
      new InputStreamReader(getClass().getResourceAsStream("/incrementalcompiler.version.properties"), java.nio.charset.StandardCharsets.UTF_8)
    )
    val VersionRegex = "^version\\=(\\d+)\\.(\\d+)\\..*".r
    reader.lines.toList.asScala.exists {
      case VersionRegex(major, minor) if major.toInt > 1 || (major.toInt == 1 && minor.toInt >= 10) => true
      case _ => false
    }
```
also works. Let me know if I should change it to that